node_modules/col-canvas/lib/api.js

```diff
@@ -143,6 +145,70 @@ var uploadFileToCanvas = module.exports.uploadFileToCanvas = function(ctx, fileP
 };
 
 /**
+ * Check for existence of upload folder in Canvas, creating if it doesn't exist.
+ *
+ * @param  {Context}        ctx                   Standard context containing the current user and the current course
+ * @param  {Function}       callback              Standard callback function
+ * @param  {Object}         callback.err          An error that occurred, if any
+ * @api private
+ */
+var checkCanvasUploadFolder = function(ctx, callback) {
+  var getFolderUrl = util.format('%s/api/v1/courses/%d/folders/by_path/%s', getCanvasBaseURI(ctx.course.canvas), ctx.course.canvas_course_id, CANVAS_UPLOAD_PATH);
+  var getFolderOpts = {
+    'url': getFolderUrl,
+    'method': 'GET',
+    'headers': {
+      'Authorization': util.format('Bearer %s', ctx.course.canvas.api_key)
+    }
+  };
+
+  request(getFolderOpts, function(err, response, body) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id}, 'Failed to check existence of Canvas upload folder');
+      return callback({'code': 500, 'msg': 'Failed to check existence of Canvas upload folder'});
+
+    } else if (response.statusCode === 200) {
+      // The folder exists; nothing more to do.
+      return callback();
+
+    } else if (response.statusCode !== 404) {
+      // We don't know what to do with this error.
+      log.error({'course': ctx.course.id, 'response': response, 'body': body}, 'Failed to check existence of Canvas upload folder');
+      return callback({'code': 500, 'msg': 'Failed to check existence of Canvas upload folder'});
```

Comment:
@paulkerschen perhaps the wording of the 404 message should differ from the message used in `if (err) { }` block. It might also be useful to log the `response.statusCode` value.